### ISSUE TITLE
feat(cache): create the .cache folder when non existant

### ISSUE THREAD

**Issue by netbix** [JanDeDobbeleer/oh-my-posh#5758]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

if the .cache folder does not exist during installation, the software will not install correctly.
Furthermore, these files are continuously created in the root:
omp.cache.9e82fb6d-f453-4846-a531-0afd015fefb1 omp.cache omp.cache.9823a730-3764-41e1-bf68-8f4e8fd38890 themes

### Theme

default theme

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash

### Log output

```Shell
[root@veeamproxy ~]# curl -s https://ohmyposh.dev/install.sh | bash -s
⚠️  Installation directory /root/.local/bin is not in your $PATH, add it using 
export PATH=$PATH:/root/.local/bin

ℹ️  Installing oh-my-posh for linux-amd64 in /root/.local/bin
⬇️  Downloading oh-my-posh from https://github.com/JanDeDobbeleer/oh-my-posh/releases/latest/download/posh-linux-amd64
🎨 Installing oh-my-posh themes in /root/themes         <<<<-----------------???

🚀 Installation complete.

You can follow the instructions at https://ohmyposh.dev/docs/installation/prompt
to setup your shell to use oh-my-posh.

If you want to use a built-in theme, you can find them in the /root/themes directory:
  oh-my-posh init {shell} --config /root/themes/{theme}.omp.json

[root@veeamproxy ~]# ls
anaconda-ks.cfg  themes
```

**Comment by JanDeDobbeleer** [comment]
> if the .cache folder does not exist during installation, the software will not install correctly.

@netbix what does this mean? I see everything did install based on the output. We look for a few places for cache data based on existing standards, and default to `HOME` when none are found by design. You can set `OMP_CACHE_DIR`, or `XDG_CACHE_HOME` based on what you prefer to define the cache directory.

**Comment by netbix** [comment]
I get it, I'm just saying it would be nicer if the .cache folder was created during the install. Furthermore, every time you enter the shell and exit, all cache files are left behind.
![immagine](https://github.com/user-attachments/assets/2d3d2ab5-8820-4dd2-8b43-9fa43999c232)

**Comment by netbix** [comment]
Thanks for your answer anyway.

**Comment by JanDeDobbeleer** [comment]
@netbix they will be cleaned after 7 days with the next major version. I noticed that as well, will be solved.

On the request, I guess that's a good idea. Didn't consider that previously, but check. Why not indeed.

**Comment by netbix** [comment]
thanks for your nice work!

**Comment by kunaaal13** [comment]
> > if the .cache folder does not exist during installation, the software will not install correctly.
> 
> @netbix what does this mean? I see everything did install based on the output. We look for a few places for cache data based on existing standards, and default to `HOME` when none are found by design. You can set `OMP_CACHE_DIR`, or `XDG_CACHE_HOME` based on what you prefer to define the cache directory.


<img width="1440" alt="Screenshot 2024-10-22 at 2 03 33 PM" src="https://github.com/user-attachments/assets/243f03cc-a34c-464e-b493-2c43481f0c91">
@JanDeDobbeleer where to add this OMP_CACHE_DIR because my root folder has become mess

**Comment by JanDeDobbeleer** [comment]
It's an environment variable, you can set to any folder you like.

**Comment by JanDeDobbeleer** [comment]
Solved in #5686, you can download the build there. Actual release pending for the next week(s).

**Comment by netbix** [comment]
> @netbix they will be cleaned after 7 days with the next major version. I noticed that as well, will be solved.
> 
> On the request, I guess that's a good idea. Didn't consider that previously, but check. Why not indeed.

Is the omp.cache file cleaning problem solved?
